api/src/main/java/com/inrupt/client/ProblemDetails.java

```diff
@@ -0,0 +1,132 @@
+    public static final String DEFAULT_TYPE = "about:blank";
+    private final URI type;
+    private final String title;
+    private final String details;
+    private final int status;
+    private final URI instance;
+
+    public ProblemDetails(
+        final URI type,
+        final String title,
+        final String details,
+        final int status,
+        final URI instance
+    ) {
+        // The `type` is not mandatory in RFC9457, so we want to set
+        // a default value here even when deserializing from JSON.
+        if (type != null) {
+            this.type = type;
+        } else {
+            this.type = URI.create(DEFAULT_TYPE);
+        }
+        this.title = title;
+        this.details = details;
+        this.status = status;
+        this.instance = instance;
+    }
+
+    public URI getType() {
+        return this.type;
+    };
+
+    public String getTitle() {
+        return this.title;
+    };
+
+    public String getDetails() {
+        return this.details;
+    };
+
+    public int getStatus() {
+        return this.status;
+    };
+
+    public URI getInstance() {
+        return this.instance;
+    };
+
+    public static ProblemDetails fromErrorResponse(
+            final int statusCode,
+            final Headers headers,
+            final byte[] body,
+            final JsonService jsonService
+    ) {
+        if (jsonService == null
+                || (headers != null && !headers.allValues("Content-Type").contains(ProblemDetails.MIME_TYPE))) {
+            return new ProblemDetails(
+                null,
+                HttpStatus.StatusMessages.getStatusMessage(statusCode),
+                null,
+                statusCode,
+                null
+            );
+        }
+        try {
+            // ProblemDetails doesn't have a default constructor, and we can't use JSON mapping annotations because
+            // the JSON service is an abstraction over JSON-B and Jackson, so we deserialize the JSON object in a Map
+            // and build the ProblemDetails from the Map values.
+            final Map<String, Object> pdData = jsonService.fromJson(
+                    new ByteArrayInputStream(body),
+                    new HashMap<String, Object>(){}.getClass().getGenericSuperclass()
+            );
+            final String title = Optional.ofNullable((String) pdData.get("title"))
+                    .orElse(HttpStatus.StatusMessages.getStatusMessage(statusCode));
+            final String details = (String) pdData.get("details");
+            final URI type = Optional.ofNullable((String) pdData.get("type"))
+                    .map(URI::create)
+                    .orElse(null);
+            final URI instance = Optional.ofNullable((String) pdData.get("instance"))
+                    .map(URI::create)
+                    .orElse(null);
+            // Note that the status code is disregarded from the body, and reused from the HTTP response directly,
```

Comment:
We could say that the mentioned requirement is for the server.
If the server decides to err, I think we should return what the server said.

Reply:
With the typed JSON deserialization, we no longer enforce the problem details status is the same as the HTTP response status, except in case of invalid status in the first place.